packages/babel-core/src/transformation/file/index.js

```diff
@@ -410,8 +412,32 @@ export default class File extends Store {
   }
 
   parse(code: string) {
+    let opts = this.opts;
+    let parseCode = parse;
+    if (opts.parserOpts.parser) {
+      parseCode = opts.parserOpts.parser;
+
+      if (typeof parseCode === "string") {
+        let dirname = opts.parserOpts.dirname || process.cwd();
+        let parser = resolve(parseCode, dirname);
+        if (parser) {
+          parseCode = require(parser).parse;
+        } else {
+          throw new Error(`Couldn't find parser ${parseCode} with "parse" method relative to directory ${dirname}`);
+        }
+      }
+
+      if (!this.parserOpts.parser) {
+        this.parserOpts.parser = {
```

Comment:
I think because the initial defaults don't have a parser (so actually it would always evaluate to true).

This is all tied to recast so maybe we need to PR that.

`parseCode = require(parser).parse;`

I copied it from https://github.com/benjamn/recast/blob/master/test/babylon.js#L31-L37

Because otherwise it uses the default parser